nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/stream/StreamPubSubProvider.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * (C) Copyright 2017 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.runtime.stream;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.function.BiConsumer;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.lib.stream.computation.Record;
+import org.nuxeo.lib.stream.log.LogAppender;
+import org.nuxeo.lib.stream.log.LogRecord;
+import org.nuxeo.lib.stream.log.LogTailer;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.pubsub.AbstractPubSubProvider;
+
+/**
+ * @since 10.1
+ */
+public class StreamPubSubProvider extends AbstractPubSubProvider {
+    private static final Log log = LogFactory.getLog(StreamPubSubProvider.class);
+
+    public static final String GROUP_PREFIX = "pub-sub";
+
+    public static final String NODE_ID_PROP = "repository.clustering.id";
+
+    protected static final Random RANDOM = new Random();
+
+    protected String logConfig;
+
+    protected String logName;
+
+    protected LogAppender<Record> appender;
+
+    protected Thread thread;
+
+    @Override
+    public void initialize(Map<String, String> options, Map<String, List<BiConsumer<String, byte[]>>> subscribers) {
+        log.debug("Initializing ");
+        super.initialize(options, subscribers);
+        logConfig = options.getOrDefault("logConfig", "default");
+        logName = options.get("logName");
```

Comment:
Please use constants (even if local to this file) for `"logConfig"`, `"logName"` and `"default"`